Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean

```diff
@@ -94,7 +94,8 @@ noncomputable def matrixDecomposition (o : HomOrthogonal s) {α β : Type} [Fint
       if h : f j = g k then z (f j) ⟨k, by simp [h]⟩ ⟨j, by simp⟩ ≫ eqToHom (by simp [h]) else 0
   left_inv z := by
     ext (j k)
-    simp only [biproduct.matrix_π, biproduct.ι_desc]
+    simp only [eqToHom_refl, Category.id_comp, Category.assoc, eqToHom_trans, Category.comp_id,
+      dite_eq_ite, biproduct.ι_matrix, biproduct.lift_π]
```

Comment:
:-(

It looks like we switched to a worse `ext` lemma. Perhaps leave a note here? We really need to deal with the `ext` regressions since mathlib3.

Reply:
The original proof did
```
    apply biproduct.hom_ext; intro j
    apply biproduct.hom_ext'; intro k
```
at the `ext j k` line, and the new one does
```
    apply biproduct.hom_ext'; intro j
    apply biproduct.hom_ext; intro k
```
here. Note that both `biproduct.hom_ext` and `biproduct.hom_ext'` are applicable as the first step, i.e. this is a non-confluent choice of ext lemma. I can investigate why the application order changed, but all other things being equal I would have expected `biproduct.hom_ext'` to be applied first because it is declared last.

To un-regress this proof you could just use `apply` for the first line and `ext` for the second.

Reply:
I found the issue, it is a bug on my part: I replaced a use of `Array.insertionSort` with `Array.qsort`, which sorts the list of ext lemmas by priority. `qsort` is not a stable sort, so they end up scrambled in the (very common!) case when all the lemmas have the same priority.

Reply:
Thanks for debugging (maybe I should have done more than simply going "🤷 ext changed"). Will you push a fix to std? in which case this PR can morph into bump to std06-01 or something

Reply:
I'll push a fix shortly.

Reply:
Pushed a fix. But keep in mind that as described above, the correct ordering here is to try `hom_ext'` first, and hence you will get the new "bad" proof. If you want the other order you should either declare `hom_ext` last or give it higher priority than `hom_ext'`.

Reply:
FWIW in lean 3 it also applies `hom_ext'` first; the original proof was
```lean
  begin
    ext j k,
    simp only [category.assoc, biproduct.lift_π, biproduct.ι_matrix],
    split_ifs,
    { simp, refl, },
    { symmetry, apply o.eq_zero h, },
  end
```